diff --git a/deps/uv/uv.gyp b/deps/uv/uv.gyp
index 7fc7e0601b..e891f53847 100644
--- a/deps/uv/uv.gyp
+++ b/deps/uv/uv.gyp
@@ -261,6 +261,7 @@
             'src/unix/procfs-exepath.c',
             'src/unix/random-getrandom.c',
             'src/unix/random-sysctl-linux.c',
+            'src/unix/random-getentropy.c',
           ],
           'link_settings': {
             'libraries': [ '-ldl' ],
